cli/tests/stake.rs

```diff
@@ -2319,3 +2320,312 @@ fn test_stake_minimum_delegation() {
+            inactive: 0,
+        }
+    );
+
+    // `stake_keypair2` should now be activating
+    assert_eq!(
+        rpc_client
+            .get_stake_activation(stake2_keypair.pubkey(), None)
+            .unwrap(),
+        RpcStakeActivation {
+            state: StakeActivationState::Activating,
+            active: 0,
+            inactive: 50_000_000_000 - rent_exempt_reserve,
+        }
+    );
+
+    // check that all the stake, save `rent_exempt_reserve`, have been moved from `stake_keypair`
+    // to `stake2_keypair`
+    check_balance!(rent_exempt_reserve, &rpc_client, &stake_keypair.pubkey());
+    check_balance!(50_000_000_000, &rpc_client, &stake2_keypair.pubkey());
+
+    // Deactivate stake2_account should fail because stake2_account is not fully activated.
+    config.signers = vec![&default_signer];
+
+    config.command = CliCommand::DeactivateStake {
+        stake_account_pubkey: stake2_keypair.pubkey(),
+        stake_authority: 0,
+        sign_only: false,
+        deactivate_delinquent: false,
+        dump_transaction_message: true,
+        blockhash_query: BlockhashQuery::default(),
+        nonce_account: None,
+        nonce_authority: 0,
+        memo: None,
+        seed: None,
+        fee_payer: 0,
+        compute_unit_price: None,
+    };
+    let deactivate_result = process_command(&config);
+    if let Err(e) = deactivate_result {
+        let s = format!("{}", e);
+        assert_eq!(
+            s,
+            "redelegated stake must be fully activated before deactivation"
+        );
+    } else {
+        unreachable!();
+    }
+
+    // `stake_keypair2` should still be Activating
+    assert_eq!(
+        rpc_client
+            .get_stake_activation(stake2_keypair.pubkey(), None)
+            .unwrap(),
+        RpcStakeActivation {
+            state: StakeActivationState::Activating,
+            active: 0,
+            inactive: 50_000_000_000 - rent_exempt_reserve,
+        }
+    );
+
+    // Withdraw from stake2 account should still fails
+    let recipient = keypair_from_seed(&[5u8; 32]).unwrap();
+    let recipient_pubkey = recipient.pubkey();
+    config.command = CliCommand::WithdrawStake {
+        stake_account_pubkey: stake2_keypair.pubkey(),
+        destination_account_pubkey: recipient_pubkey,
+        amount: SpendAmount::Some(50_000_000_000),
+        withdraw_authority: 0,
+        custodian: None,
+        sign_only: false,
+        dump_transaction_message: false,
+        blockhash_query: BlockhashQuery::All(blockhash_query::Source::Cluster),
+        nonce_account: None,
+        nonce_authority: 0,
+        memo: None,
+        seed: None,
+        fee_payer: 0,
+        compute_unit_price: None,
+    };
+
+    let withdraw_result = process_command(&config);
+    if let Err(e) = withdraw_result {
+        let s = format!("{}", e);
+        assert!(s.contains("insufficient funds for instruction"));
+    } else {
+        unreachable!();
+    }
```

Comment:
Can you add some more logic to wait one epoch and then deactivate properly?

Reply:
done